search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt

```diff
@@ -129,21 +132,29 @@ class EntityEventListenerService(
     private fun handleAttributeUpdate(
         entityId: URI,
         expandedAttributeName: String,
+        datasetId: URI?,
         attributeValuesNode: JsonNode,
         updatedEntity: String,
         contexts: List<String>
     ) {
         // TODO add missing checks:
         //  - existence of temporal entity attribute
-        when (val extractedAttributeMetadata = toTemporalAttributeMedata(attributeValuesNode)) {
+        val compactedJsonLdEntity = addContextsToEntity(JsonUtils.deserializeObject(updatedEntity), contexts)
```

Comment:
that is extra processing for all received attributes, even those who are not temporal (that's the majority of them). it would be nice to return early if the attribute is not temporal.